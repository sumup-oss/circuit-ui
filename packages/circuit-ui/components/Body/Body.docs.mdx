import { Status, Props, Story } from '../../../../.storybook/components';

# Body

<Status.Stable />

Body is used to present the core textual content for our users. They are contained within headlines and subheadlines
and have five different variants `highlight`, `quote`, `success`, `error` and `subtle` to tailor it according to the content we are
presenting.

<Story id="typography-body--base" />
<Props />

## Usage guidelines

- **Do** use the `one` size as the default for body text
- **Do** use spacing to better organize textual information, such as different paragraphs
- **Do** use different variants of the body component

### Sizes

<Story id="typography-body--sizes" />

### Variants

<Story id="typography-body--variants" />
