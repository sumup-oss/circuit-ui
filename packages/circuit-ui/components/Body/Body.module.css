/* Sizes */

.l {
  font-size: var(--cui-typography-body-l-font-size);
  line-height: var(--cui-typography-body-l-line-height);
  letter-spacing: var(--cui-typography-body-l-letter-spacing);
}

.m {
  font-size: var(--cui-typography-body-m-font-size);
  line-height: var(--cui-typography-body-m-line-height);
  letter-spacing: var(--cui-typography-body-m-letter-spacing);
}

.s {
  font-size: var(--cui-typography-body-s-font-size);
  line-height: var(--cui-typography-body-s-line-height);
  letter-spacing: var(--cui-typography-body-s-letter-spacing);
}

/* Weights */

.regular {
  font-weight: var(--cui-font-weight-regular);
}

.bold {
  font-weight: var(--cui-font-weight-bold);
}

/* Colors */

.normal {
  color: var(--cui-fg-normal);
}

.subtle {
  color: var(--cui-fg-subtle);
}

.placeholder {
  color: var(--cui-fg-placeholder);
}

.on-strong {
  color: var(--cui-fg-on-strong);
}

.on-strong-subtle {
  color: var(--cui-fg-on-strong-subtle);
}

.accent {
  color: var(--cui-fg-accent);
}

.success {
  color: var(--cui-fg-success);
}

.warning {
  color: var(--cui-fg-warning);
}

.danger {
  color: var(--cui-fg-danger);
}

.promo {
  color: var(--cui-fg-promo);
}

/* Variants */

.highlight,
strong {
  font-weight: var(--cui-font-weight-bold);
}

.quote,
blockquote {
  padding-left: var(--cui-spacings-kilo);
  font-style: italic;
  border-left: var(--cui-border-width-mega) solid var(--cui-border-accent);
}

.confirm {
  color: var(--cui-fg-success);
}

.alert {
  color: var(--cui-fg-danger);
}
