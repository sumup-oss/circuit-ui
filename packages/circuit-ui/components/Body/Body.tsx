/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { forwardRef, HTMLAttributes, Ref } from 'react';
import { css } from '@emotion/react';
import isPropValid from '@emotion/is-prop-valid';

import styled, { StyleProps } from '../../styles/styled';
import { AsPropType, EmotionAsPropType } from '../../types/prop-types';

type Size = 'one' | 'two';
type Variant = 'highlight' | 'quote' | 'confirm' | 'alert' | 'subtle';

export interface BodyProps extends HTMLAttributes<HTMLParagraphElement> {
  /**
   * Choose from 2 font sizes. Default `one`.
   */
  size?: Size;
  /**
   * Choose from style variants.
   */
  variant?: Variant;
  /**
   * Render the text using any HTML element.
   */
  as?: AsPropType;
  /**
   * The ref to the HTML DOM element.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref?: Ref<any>;
}

const baseStyles = ({ theme }: StyleProps) => css`
  font-weight: ${theme.fontWeight.regular};
`;

const sizeStyles = ({ theme, size = 'one' }: BodyProps & StyleProps) => css`
  font-size: ${theme.typography.body[size].fontSize};
  line-height: ${theme.typography.body[size].lineHeight};
`;

const variantStyles = ({ theme, variant }: BodyProps & StyleProps) => {
  // TODO: Align variant names with token names in the next major.
  switch (variant) {
    case 'highlight': {
      return css`
        font-weight: ${theme.fontWeight.bold};
      `;
    }
    case 'quote': {
      return css`
        font-style: italic;
        padding-left: ${theme.spacings.kilo};
        border-left: ${theme.borderWidth.mega} solid var(--cui-border-accent);
      `;
    }
    case 'confirm': {
      return css`
        color: var(--cui-fg-success);
      `;
    }
    case 'alert': {
      return css`
        color: var(--cui-fg-danger);
      `;
    }
    case 'subtle': {
      return css`
        color: var(--cui-fg-subtle);
      `;
    }
    default: {
      return null;
    }
  }
};

const StyledBody = styled('p', {
  shouldForwardProp: (prop) => isPropValid(prop) && prop !== 'size',
})<BodyProps>(baseStyles, sizeStyles, variantStyles);

function getHTMLElement(variant?: Variant): AsPropType {
  if (variant === 'highlight') {
    return 'strong';
  }
  if (variant === 'quote') {
    return 'blockquote';
  }
  return 'p';
}

/**
 * The Body component is used to present the core textual content
 * to our users.
 */
export const Body = forwardRef((props: BodyProps, ref?: BodyProps['ref']) => {
  const as = props.as || getHTMLElement(props.variant);
  return <StyledBody {...props} ref={ref} as={as as EmotionAsPropType} />;
});

Body.displayName = 'Body';
