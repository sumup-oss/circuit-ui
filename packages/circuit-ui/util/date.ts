/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Temporal } from 'temporal-polyfill';
import { formatDateTime } from '@sumup-oss/intl';

import type { Locale } from './i18n.js';

export type FirstDayOfWeek = 1 | 7;
export type DaysInWeek = number;
export type PlainDateRange =
  | []
  | [Temporal.PlainDate]
  | [Temporal.PlainDate, Temporal.PlainDate];

// ISO 8601 timestamps only support positive 4-digit years
export const MIN_YEAR = 1;
export const MAX_YEAR = 9999;

export const MIN_MONTH = 1;
export const MAX_MONTH = 12;

export const MIN_DAY = 1;
// MAX_DAY is not constant as it depends on the year and month

export function getTodaysDate() {
  return Temporal.Now.plainDateISO();
}

export function isPlainDate(date: unknown): date is Temporal.PlainDate {
  return date instanceof Temporal.PlainDate;
}

export function toPlainDate(date?: string): Temporal.PlainDate | undefined {
  if (!date) {
    return undefined;
  }
  try {
    return Temporal.PlainDate.from(date);
  } catch {
    return undefined;
  }
}

export function sortDateRange<T extends PlainDateRange>(dateRange: T): T {
  return dateRange.sort((a, b) => Temporal.PlainDate.compare(a, b)) as T;
}

export function clampDate(
  date: Temporal.PlainDate,
  minDate?: Temporal.PlainDate | null,
  maxDate?: Temporal.PlainDate | null,
): Temporal.PlainDate {
  if (minDate && Temporal.PlainDate.compare(date, minDate) < 0) {
    return minDate;
  }
  if (maxDate && Temporal.PlainDate.compare(date, maxDate) > 0) {
    return maxDate;
  }
  return date;
}

export function getFirstDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  const diff =
    (date.dayOfWeek < firstDayOfWeek ? date.daysInWeek : 0) +
    date.dayOfWeek -
    firstDayOfWeek;
  return date.subtract({ days: diff });
}

export function getLastDateOfWeek(
  date: Temporal.PlainDate,
  firstDayOfWeek: FirstDayOfWeek,
): Temporal.PlainDate {
  return getFirstDateOfWeek(date, firstDayOfWeek).add({
    days: date.daysInWeek - 1,
  });
}

export function getMonthName(month: number, locale?: Locale) {
  // The year can be arbitrary since the month names are the same every year
  const yearMonth = new Temporal.PlainYearMonth(2000, month, 'gregory');
  return formatDateTime(yearMonth, locale, { month: 'long' });
}
