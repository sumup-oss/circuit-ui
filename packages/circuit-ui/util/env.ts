/**
 * Copyright 2024, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export function getEnvVariable(name: string) {
  if (name === 'NODE_ENV') {
    // Some bundlers have special logic for `process.env.NODE_ENV` which
    // relies on it being written as a continuous string. Destructuring or
    // dynamically accessing `NODE_ENV` can break this logic.
    throw new Error('Do not dynamically access NODE_ENV');
  }

  if (typeof process !== 'undefined') {
    return process.env?.[name];
  }

  return undefined;
}
