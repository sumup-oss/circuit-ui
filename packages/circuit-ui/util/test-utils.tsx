/**
 * Copyright 2020, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FunctionComponent, ReactElement } from 'react';
import { renderToStaticMarkup } from 'react-dom/server';
import '@testing-library/jest-dom/extend-expect';
import { configureAxe } from 'jest-axe';
import { render as renderTest, RenderResult } from '@testing-library/react';
import { renderHook, act as actHook } from '@testing-library/react-hooks';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'emotion-theming';
import { light } from '@sumup/design-tokens';

import {
  ComponentsContext,
  defaultComponents,
} from '../components/ComponentsContext/ComponentsContext';

export * from '@testing-library/react';

export type RenderFn<T = any> = (component: ReactElement, ...rest: any) => T;

const WithProviders: FunctionComponent = ({ children }) => (
  <ComponentsContext.Provider value={defaultComponents}>
    <ThemeProvider theme={light}>{children}</ThemeProvider>
  </ComponentsContext.Provider>
);

const render: RenderFn<RenderResult> = (component, options) =>
  renderTest(component, { wrapper: WithProviders, ...options });
const renderToHtml: RenderFn<string> = (component) =>
  renderToStaticMarkup(<WithProviders>{component}</WithProviders>);
const create = (...args: Parameters<RenderFn<RenderResult>>) => {
  const { container } = render(...args);
  return container.children.length > 1
    ? container.children
    : container.firstChild;
};

const axe = configureAxe({
  rules: {
    // disabled landmark rules when testing isolated components.
    region: { enabled: false },
  },
});

export { create, render, renderToHtml, renderHook, actHook, userEvent, axe };
