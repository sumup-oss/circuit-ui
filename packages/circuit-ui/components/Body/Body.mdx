import { Meta, Status, Props, Story } from '../../../../.storybook/components';
import * as Stories from './Body.stories';

<Meta of={Stories} />

# Body

<Status.Stable />

The Body component is used to present content to our users.

<Story of={Stories.Base} />
<Props />

## Usage guidelines

## Component variations

### Sizes

The Body component comes in two sizes. Use the default `one` size in most cases. Consider using the [BodyLarge component](Typography/BodyLarge) for large typography in specific cases.

<Story of={Stories.Sizes} />

### Variants

The Body component accepts five different variants—`highlight`, `quote`, `confirm`, `alert` and `subtle`—to tailor it according to the content we are presenting.

Different variants will render different HTML elements by default:

| Variant     | Default element |
| ----------- | --------------- |
| _default_   | `p`             |
| `highlight` | `strong`        |
| `quote`     | `blockquote`    |
| `confirm`   | `p`             |
| `alert`     | `p`             |
| `subtle`    | `p`             |

Use the `as` prop to render other HTML elements, for example for rendering a highlighted paragraph (`<Body variant="highlight" as="p" />`).

<Story of={Stories.Variants} />

**Important**: the `confirm` and `alert` variants only change the text's color. Keep in mind that this is not sufficient to make them accessible. Refer to the Accessibility section below, especially "Do not rely on color alone" and "Make status messages accessible to screen readers using live regions".

---

## Accessibility

### Best practices

#### Break text up into sections

In order to make content easier to digest (especially on content-heavy pages such as articles or landing pages), break text up into sections.

- Use semantic HTML elements and headings (use the [Headline](Typography/Headline) component) to build sections into your markup
- Add spacing between paragraphs and sections for sighted users (use the [`spacing()` style mixin](Features/Style-Mixins/Spacing))

This is beneficial to everyone, but critical for users with cognitive, language or learning disabilities (such as dyslexia or ADHD).

#### Write simple copy

Similarly to how breaking text up into sections helps users _parse_ content, writing concise and simple copy helps users _understand_ content.

- Avoid the figurative use of words, or specialized words ([3.1.3: Unusual Words](https://www.w3.org/WAI/WCAG21/Understanding/unusual-words))
- Allow users to access the expanded form of abbreviations ([3.1.4: Abbreviations](https://www.w3.org/WAI/WCAG21/Understanding/abbreviations))
- Write content as clearly and simply as possible ([3.1.5: Reading Level](https://www.w3.org/WAI/WCAG21/Understanding/reading-level))

_Note: the success criteria referenced above are intended for AAA conformance. Though they are not requirements, they are a best practice and following them will improve the accessibility of our content._

#### Complement text with images

Images, graphs, or other illustrations can help users contextualize and/or understand a piece of content.

Visuals should usually complement text—and not replace it—unless an excellent description is provided as alternative text.

#### Translate content

Translate content intended for a linguistically diverse audience, and make it easy for users to change the page's language.

Bear in mind that machine translation is often inaccurate: prefer professional translation in order to give the best possible experience to users from diverse linguistic backgrounds.

#### Do not rely on color alone

Color alone is not sufficient to differentiate text from the surrounding content ([1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)).

This is especially relevant when using the `confirm` and `alert` variants.

For example, imagine a list of good and bad accessibility practices. It is not enough to use the `confirm` (green) variant for good practices and the `alert` (red) variant for bad practices. Instead, use additional visual cues such as icons (with an accessible label), or break up the list into two separate ones with explicit labeling.

#### Make status messages accessible to screen readers using live regions

When a change in content is not given focus (typically a message being added to the DOM to reflect a status update), the change needs to be announced to screen readers using a [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) ([4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html), [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification)).

For example, a form submission fails and a message saying "This username is already taken." using `<Body variant="alert" />` appears. Without a live region, screen reader users would have no way of knowing that something happened and that they need to make changes to the form before submitting again. They will either have to step through the DOM to try and find the relevant error message, or they will give up and close the page.

Live regions can be tricky to work with, therefore we recommend using existing components such as the [NotificationInline](Notifications/NotificationInline) or the [NotificationToast](Notifications/NotificationToast) instead of implementing them from scratch.

### Resources

#### Visualizing document structure

Use a tool like [Wave](https://wave.webaim.org/) to extract structure from a page (in Wave, you'll find the page structure under the _Structure_ tab). Verify that your copy is accurately grouped and labeled by headings.

#### Verify your UI without color

Simulate vision deficiencies or desaturate your page, and verify that visual cues beyond color are helping users understand your copy.

- [Simulate vision deficiencies using Firefox](https://developer.mozilla.org/en-US/docs/Tools/Accessibility_inspector/Simulation)
- [Simulate vision deficiencies using Chrome](https://developer.chrome.com/blog/new-in-devtools-83/#vision-deficiencies)
- Desaturate a page using [Wave](https://wave.webaim.org/) (under Contrast, "Desaturate page")

#### Test your page using a screen reader

[Test your page using a screen reader](https://webaim.org/articles/screenreader_testing/) like JAWS (Windows), NVDA (Windows, Linux) or VoiceOver (macOS).

This is particularly valuable for highly dynamic content, for example using live regions.

#### Further reading

- [Writing for Web Accessibility](https://www.w3.org/WAI/tips/writing/) (w3.org)
- [Cognitive Disabilities](https://webaim.org/articles/cognitive/) (WebAIM)

#### Related WCAG success criteria

- 1.4.1: [Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html)
- 3.1.3: [Unusual Words](https://www.w3.org/WAI/WCAG21/Understanding/unusual-words) (AAA)
- 3.1.4: [Abbreviations](https://www.w3.org/WAI/WCAG21/Understanding/abbreviations) (AAA)
- 3.1.5: [Reading Level](https://www.w3.org/WAI/WCAG21/Understanding/reading-level) (AAA)
- 3.3.1: [Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification)
- 4.1.3: [Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
