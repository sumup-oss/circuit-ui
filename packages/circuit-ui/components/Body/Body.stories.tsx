/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import docs from './Body.docs.mdx';
import { BodyProps } from './Body';

import Body from '.';

const content =
  'An electronic circuit is composed of individual electronic components, such as resistors, transistors, capacitors, inductors and diodes, connected by conductive wires or traces through which electric current can flow.';

export default {
  title: 'Typography/Body',
  component: Body,
  parameters: {
    docs: { page: docs },
  },
  argTypes: {
    as: { control: 'text' },
  },
};

export const Base = (args: BodyProps) => <Body {...args}>{content}</Body>;

const sizes = ['one', 'two'] as const;

export const Sizes = (args: BodyProps) =>
  sizes.map((s) => (
    <Body key={s} {...args} size={s}>
      This is a body {s}. {content}
    </Body>
  ));

const variants = ['highlight', 'quote', 'confirm', 'alert', 'subtle'] as const;

export const Variants = (args: BodyProps) =>
  variants.map((variant) => (
    <Body key={variant} {...args} variant={variant}>
      This is a {variant} body
    </Body>
  ));
