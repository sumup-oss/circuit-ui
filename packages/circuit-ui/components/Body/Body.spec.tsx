/**
 * Copyright 2019, SumUp Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { create, renderToHtml, axe, render } from '../../util/test-utils';

import { Body, BodyProps } from './Body';

describe('Body', () => {
  /**
   * Style tests.
   */
  it('should render with default styles', () => {
    const actual = create(<Body>Body</Body>);
    expect(actual).toMatchSnapshot();
  });

  const sizes: BodyProps['size'][] = ['one', 'two'];
  it.each(sizes)('should render with size "%s"', (size) => {
    const actual = create(<Body size={size}>{size} Body</Body>);
    expect(actual).toMatchSnapshot();
  });

  const variants = [
    'highlight',
    'quote',
    'confirm',
    'alert',
    'subtle',
  ] as BodyProps['variant'][];
  it.each(variants)('should render as a "%s" variant', (variant) => {
    const actual = create(<Body variant={variant}>{variant} Body</Body>);
    expect(actual).toMatchSnapshot();
  });

  /**
   * Logic tests.
   */
  const elements = ['p', 'article', 'div'] as const;
  it.each(elements)('should render as a "%s" element', (as) => {
    const { container } = render(<Body as={as}>{as} Body</Body>);
    const actual = container.querySelector(as);
    expect(actual).toBeVisible();
  });

  it('should render the "highlight" variant as a "strong" element', () => {
    const { container } = render(<Body variant="highlight">Highlight</Body>);
    const actual = container.querySelector('strong');
    expect(actual).toBeVisible();
  });

  it('should render the "quote" variant as a "blockquote" element', () => {
    const { container } = render(<Body variant="quote">Quote</Body>);
    const actual = container.querySelector('blockquote');
    expect(actual).toBeVisible();
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<Body>Body</Body>);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
