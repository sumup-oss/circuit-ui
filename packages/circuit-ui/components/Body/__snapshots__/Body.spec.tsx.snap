// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render as a "alert" variant 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #D23F47;
}

<p
  class="circuit-0"
>
  alert
   Body
</p>
`;

exports[`Body should render as a "confirm" variant 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #138849;
}

<p
  class="circuit-0"
>
  confirm
   Body
</p>
`;

exports[`Body should render as a "highlight" variant 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 700;
}

<strong
  class="circuit-0"
>
  highlight
   Body
</strong>
`;

exports[`Body should render as a "quote" variant 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  font-style: italic;
  padding-left: 12px;
  border-left: 2px solid #3063E9;
}

<blockquote
  class="circuit-0"
>
  quote
   Body
</blockquote>
`;

exports[`Body should render as a "subtle" variant 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #666;
}

<p
  class="circuit-0"
>
  subtle
   Body
</p>
`;

exports[`Body should render with default styles 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  class="circuit-0"
>
  Body
</p>
`;

exports[`Body should render with size "one" 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5rem;
}

<p
  class="circuit-0"
>
  one
   Body
</p>
`;

exports[`Body should render with size "two" 1`] = `
.circuit-0 {
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.25rem;
}

<p
  class="circuit-0"
>
  two
   Body
</p>
`;
